schema/make_schema.py

```diff
@@ -147,14 +156,16 @@ def format_param(param: click.core.Option) -> SamCliParameterSchema:
     """
     if not param:
         raise SchemaGenerationException("Expected to format a parameter that doesn't exist")
-    if not param.type.name:
-        raise SchemaGenerationException(f"Parameter {param} passed without a type")
+    if not param.type.name and param.name not in EXCEPTION_OPTION_CLASS_MAPPER.keys():
+        raise SchemaGenerationException(f"Parameter {param} passed without a type:\n{param.type}")
 
     param_type = param.type.name.lower()
-    formatted_param_type = ""
+    formatted_param_type: Any = None
```

Comment:
Since this type can now take the type of a list, would it be valid for the JSON schema to have a single element list (eg. `["string"]`)? If a single element list is valid, then we might just be able to have things as lists of type(s). This way the `items` if check is more simple.